rust/passwords/src/sha512crypt.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2025 Arista Networks, Inc.
+// Use of this source code is governed by the Apache License 2.0
+// that can be found in the LICENSE file.
+
+use sha_crypt::{CryptError, Sha512Params, sha512_crypt_b64};
+
+pub fn sha512_crypt(password: String, salt: String) -> Result<String, CryptError> {
```

Comment:
change this to &str instead of String.

Reply:
enum with CryptError / InvalidSaltError 
derive_more for From.

return Sha512Error